api/v1alpha1/nutanix_clusterconfig_types.go

```diff
@@ -76,3 +84,39 @@ func (s NutanixPrismCentralEndpointSpec) ParseURL() (string, int32, error) {
 
 	return hostname, int32(port), nil
 }
+
+// NutanixFailureDomains is a list of FDs.
+type NutanixFailureDomains []NutanixFailureDomain
+
+// NutanixFailureDomain configures failure domain information for Nutanix.
+type NutanixFailureDomain struct {
+	// name defines the unique name of a failure domain.
+	// Name is required and must be at most 64 characters in length.
+	// It must consist of only lower case alphanumeric characters and hyphens (-).
+	// It must start and end with an alphanumeric character.
+	// This value is arbitrary and is used to identify the failure domain within the platform.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:Pattern=`[a-z0-9]([-a-z0-9]*[a-z0-9])?`
+	Name string `json:"name"`
+
+	// cluster is to identify the cluster (the Prism Element under management of the Prism Central),
+	// in which the Machine's VM will be created. The cluster identifier (uuid or name) can be obtained
+	// from the Prism Central console or using the prism_central API.
+	// +kubebuilder:validation:Required
+	Cluster NutanixResourceIdentifier `json:"cluster"`
+
+	// subnets holds a list of identifiers (one or more) of the cluster's network subnets
+	// for the Machine's VM to connect to. The subnet identifiers (uuid or name) can be
+	// obtained from the Prism Central console or using the prism_central API.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	// +listType=map
+	// +listMapKey=type
```

Comment:
IIUC we want the name to be the unique key?
```suggestion
	// +listMapKey=name
```

Reply:
user can provide uuid or name. hence type is the only fields which will be available. also this is a list so why does it need to be unique?

Reply:
The value of from `listMapKey` is used when doing server side applies https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy, IIUC having it `listMapKey= type` would be that any (or some) elements that has the same type would be replaced instead of appended?

We probably don't want these at all here.

Reply:
`listMapKey` isn't supported for ClusterClass variables iirc.

Reply:
ok, will file a tracker to fix in capx as well https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/blob/v1.4.0-alpha.2/api/v1beta1/nutanixcluster_types.go#L132

Reply:
@deepakm-ntnx That's different because that is a real CRD and as such it is supported by the API server.

Reply:
Having said that, we should definitely re-evaluate what the key means in this context in CAPX CRDs at some point.